components/external-services-mock/internal/formationnotification/handler.go

```diff
@@ -671,16 +671,39 @@ func (h *Handler) AsyncFail(writer http.ResponseWriter, r *http.Request) {
 			state = DeleteErrorAssignmentState
 		}
 
-		if err := h.executeFormationAssignmentStatusUpdateRequest(client, correlationID, state, config, formationID, formationAssignmentID); err != nil {
+		if err := h.executeFormationAssignmentStatusUpdateRequest(client, correlationID, state, &config, formationID, formationAssignmentID); err != nil {
 			log.C(ctx).Errorf("while executing status update request: %s", err.Error())
 		}
 	}
 	config := "test error"
 	h.asyncFAResponse(ctx, writer, r, operation, config, responseFunc)
 }
 
+// AsyncFailNoError handles asynchronous formation assignment notification requests for both Assign and Unassign operations by failing and setting error states.
+func (h *Handler) AsyncFailNoError(writer http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	operation := Assign
+	if r.Method == http.MethodPatch {
+		operation = Assign
```

Comment:
We could remove the assign/patch `if` as the assign operation is the default value from the above and leave only one `if` statement for the delete/unassign?